src/pcre2_compile.c

```diff
@@ -6405,76 +6407,86 @@ for (;; pptr++)
         range += 2;
         }
 
-      if ((xclass_props & XCLASS_HIGH_ANY) != 0)
+      if (cranges->char_lists_size > 0)
         {
-        PCRE2_ASSERT(range + 2 == end && range[0] <= 256 &&
-          range[1] >= GET_MAX_CHAR_VALUE(utf));
-        should_flip_negation = TRUE;
-        range = end;
+        /* The cranges structure is still used and freed later. */
+        PCRE2_ASSERT((xclass_props & XCLASS_HIGH_ANY) == 0);
+        xclass_props |= XCLASS_REQUIRED | XCLASS_HAS_CHAR_LISTS;
         }
-
-      while (range < end)
+      else
         {
-        uint32_t range_start = range[0];
-        uint32_t range_end = range[1];
+        if ((xclass_props & XCLASS_HIGH_ANY) != 0)
+          {
+          PCRE2_ASSERT(range + 2 == end && range[0] <= 256 &&
```

Comment:
isn't this supposed to be `< 256` here?

Reply:
No. The first 256 (0..255) characters are handled by OP_CLASS/OP_NCLASS, and characters >= 256 are all match or don't.

Reply:
makes sense now, so should we ship ip?

Reply:
What does "ip" means? it?

Reply:
yes, sorry for the typo

Reply:
had prepared 35f47c52aa62c7a4eb1ac31d82583e6e408dbc8e with a few more related const that might be nice to include, but nothing to object.

Reply:
I have added them. I land the patch then.